### DISCUSSION TITLE
How should I use MONAI's HausdorffDistanceMetric function for UNETR segmentaton result?

### DISCUSSION THREAD

**Discussion by AustinYuAo** [Project-MONAI/MONAI#4619] [Q&A]
Hi, I want to use MONAI's HausdorffDistanceMetric function in UNETR repository's test.py.
As shown below, I have tried to add codes in the test.py. 
But it didn't work,what should I do? 
Thank you!!
from monai.metrics import HausdorffDistanceMetric
 hd_metric = HausdorffDistanceMetric(include_background=False, distance_metric="euclidean", percentile=None, directed=True)

    with torch.no_grad():
        dice_list_case = []
        for i, batch in enumerate(val_loader):
            val_inputs, val_labels = (batch["image"].cuda(), batch["label"].cuda())
            img_name = batch['image_meta_dict']['filename_or_obj'][0].split('/')[-1]
            print("Inference on case {}".format(img_name))
            val_outputs = sliding_window_inference(val_inputs,
                                                   (96, 96, 96),
                                                   4,
                                                   model,
                                                   overlap=args.infer_overlap)
            val_outputs = torch.softmax(val_outputs, 1).cpu().numpy()
            val_outputs = np.argmax(val_outputs, axis=1).astype(np.uint8)
            val_labels = val_labels.cpu().numpy()[:, 0, :, :, :]
            dice_list_sub = []
            for i in range(1, 14):
                organ_Dice = dice(val_outputs[0] == i, val_labels[0] == i)
                organ_HD =hd_metric.compute_hausdorff_distance(val_outputs[0] == i, val_labels[0] == i)###hausdorff distance!!!!
                dice_list_sub.append(organ_Dice)
            mean_dice = np.mean(dice_list_sub)
            print("Mean Organ Dice: {}".format(mean_dice))
            dice_list_case.append(mean_dice)
        print("Overall Mean Dice: {}".format(np.mean(dice_list_case)))

**Comment by wyli** [comment]
Could you please elaborate 'it did not work'? What is the expected and the observations?
